app/scripts/directives/fromFileDialog.js

```diff
@@ -0,0 +1,93 @@
+'use strict';
+
+angular.module('openshiftConsole').component('fromFileDialog', {
+  controller: [
+    '$scope',
+    '$timeout',
+    'DataService',
+    FromFileDialog
+  ],
+  controllerAs: '$ctrl',
+  bindings: {
+    visible: '<',
+    project: '<', //handle create project optionally
+    context: '<',
+    onDialogClosed: '&'
+  },
+  templateUrl: 'views/directives/from-file-dialog.html'
+});
+
+function FromFileDialog($scope, $timeout, DataService) {
+  var ctrl = this;
+
+  ctrl.$onInit = function() {
+    ctrl.alerts = {};
+    ctrl.loginBaseUrl = DataService.openshiftAPIBaseUrl();
+  };
+
+  function getIconClass() {
+    var icon = _.get(ctrl, 'template.metadata.annotations.iconClass', 'fa fa-cubes');
```

Comment:
Yeah this is consistent with the catalog and this dialog, for now, doesn't show up in the old experience. I'll open an issue.